File: src/TrafficLightSystem/StateRecord.java
/**
 * Javier Bastande
 */
package trafficLightSystem;


/**
 * 
 */
public class StateRecord {

	// vars
	private static int tlsID;
	private static String state;
	private static StateRecord instance;
	
	// Private constructor
	public StateRecord(int tlsID, String newState){
		this.tlsID = tlsID;
		this.state = newState;

	}
	
	// getters


	/**
	 * @return the tlsID
	 */
	public int getTLSID() {
		return tlsID;
	}

	
	/**
	 * 
	 * Get state
	 * 
	 * @return the state
	 */
	public String getState() {
		return state;
	}

	
	// setters
	
	/**
	 * @param tlsID the tlsID to set
	 */
	public void setTLSID(int tlsID) {
		StateRecord.tlsID = tlsID;
	}

	/**
	 * Set state
	 * 
	 * @param state the state to set
	 */
	public void setState(String state) {
		StateRecord.state = state;
	}
	
	
}

File: src/services/trafficLightSystem/StateRecord.js
/**
 * Javier Bastande
 */
class StateRecord {
    // vars
    static tlsID;
    static state;
    constructor(tlsID, newState) {
        StateRecord.tlsID = tlsID;
        StateRecord.state = newState;
    }

    // getters

    /**
     * @returns the tlsID
     */
    getTLSID() {
        return StateRecord.tlsID;
    }

    /**
     * Get state
     * @returns the state
     */
    getState() {
        return StateRecord.state;
    }

    // setters

    /**
     * @param tlsID the tlsID to set
     */
    setTLSID(tlsID) {
        StateRecord.tlsID = tlsID;
    }

    /**
     * Set state
     * @param state the state to set
     */
    setState(state) {
        StateRecord.state = state;
    }
}
